ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/MLSdkAsyncHttpResponseHandler.java

```diff
@@ -0,0 +1,194 @@
+import org.opensearch.ml.common.utils.StringUtils;
+import org.opensearch.script.ScriptService;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.google.gson.Gson;
+
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.http.SdkHttpResponse;
+import software.amazon.awssdk.http.async.SdkAsyncHttpResponseHandler;
+
+@Log4j2
+public class MLSdkAsyncHttpResponseHandler implements SdkAsyncHttpResponseHandler {
+    @Getter
+    private Integer statusCode;
+    @Getter
+    private final StringBuilder responseBody = new StringBuilder();
+
+    private final ExecutionContext executionContext;
+
+    private final ActionListener<List<ModelTensors>> actionListener;
+
+    private final Map<String, String> parameters;
+
+    private final Map<Integer, ModelTensors> tensorOutputs;
+
+    private final Connector connector;
+
+    private final ScriptService scriptService;
+
+    private final MLGuard mlGuard;
+
+    private final static Gson GSON = StringUtils.gson;
+
+    public MLSdkAsyncHttpResponseHandler(
+        ExecutionContext executionContext,
+        ActionListener<List<ModelTensors>> actionListener,
+        Map<String, String> parameters,
+        Map<Integer, ModelTensors> tensorOutputs,
+        Connector connector,
+        ScriptService scriptService,
+        MLGuard mlGuard
+    ) {
+        this.executionContext = executionContext;
+        this.actionListener = actionListener;
+        this.parameters = parameters;
+        this.tensorOutputs = tensorOutputs;
+        this.connector = connector;
+        this.scriptService = scriptService;
+        this.mlGuard = mlGuard;
+    }
+
+    @Override
+    public void onHeaders(SdkHttpResponse response) {
+        SdkHttpFullResponse sdkResponse = (SdkHttpFullResponse) response;
+        log.debug("received response headers: " + sdkResponse.headers());
+        this.statusCode = sdkResponse.statusCode();
+    }
+
+    @Override
+    public void onStream(Publisher<ByteBuffer> stream) {
+        stream.subscribe(new MLResponseSubscriber());
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        log.error(error.getMessage(), error);
+        RestStatus status = (statusCode == null) ? RestStatus.INTERNAL_SERVER_ERROR : RestStatus.fromCode(statusCode);
+        String errorMessage = "Error communicating with remote model: " + error.getMessage();
+        actionListener.onFailure(new OpenSearchStatusException(errorMessage, status));
+    }
+
+    private void processResponse(
+        Integer statusCode,
+        String body,
+        Map<String, String> parameters,
+        Map<Integer, ModelTensors> tensorOutputs
+    ) {
+        if (Strings.isBlank(body)) {
+            log.error("Remote model response body is empty!");
+            if (executionContext.getExceptionHolder().get() == null)
```

Comment:
How about add `{}` to make code more readable ? https://stackoverflow.com/questions/20116639/java-one-line-if-statement